src/trustymail_csv2mongo.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/local/bin/python3
+
+"""Populate MongoDB from a sorted CSV file as output by domain scan."""
+
+# Standard Python Libraries
+import csv
+import datetime
+
+# Third-Party Libraries
+from mongo_db_from_config import db_from_config
+
+DB_CONFIG_FILE = "/run/secrets/scan_write_creds.yml"
+INCLUDE_DATA_DIR = "/home/saver/include/"
+SHARED_DATA_DIR = "/home/saver/shared/"
+
+AGENCIES_FILE = INCLUDE_DATA_DIR + "agencies.csv"
+
+CURRENT_FEDERAL_FILE = SHARED_DATA_DIR + "artifacts/current-federal_modified.csv"
+UNIQUE_AGENCIES_FILE = SHARED_DATA_DIR + "artifacts/unique-agencies.csv"
+CLEAN_CURRENT_FEDERAL_FILE = SHARED_DATA_DIR + "artifacts/clean-current-federal.csv"
+
+TRUSTYMAIL_RESULTS_FILE = SHARED_DATA_DIR + "artifacts/results/trustymail.csv"
+
+
+class Domainagency:
+    """A data container for domain and agency."""
+
+    def __init__(self, domain, agency):
+        """Initialize the container.
+
+        :param domain: The domain object
+        :param agency: The agency object
+        """
+        self.domain = domain
+        self.agency = agency
+
+
+def main():
+    """Save the trustymail data."""
+    opened_files = open_csv_files()
+    store_data(opened_files[0], opened_files[1], DB_CONFIG_FILE)
+
+
+def open_csv_files():
+    """Return information about federal agencies from CSV files.
+
+    :return: A cleaned up version of current federal and a dict of
+    agency IDs keyed by agency name
+    """
+    agencies = open(AGENCIES_FILE)
+    current_federal = open(CURRENT_FEDERAL_FILE)
+
+    unique_agencies = open(UNIQUE_AGENCIES_FILE, "w+")
+    # Create a writer so we have a list of our unique agencies
+    writer = csv.writer(unique_agencies, delimiter=",")
+
+    # Get the cleaned current-federal
+    clean_federal = []
+    unique = set()
+    for row in csv.DictReader(current_federal):
+        domain = row["Domain Name"]
+        agency = (
+            row["Agency"]
+            .replace("&", "and")
+            .replace("/", " ")
+            .replace("U. S.", "U.S.")
+            .replace(",", "")
+        )
+
+        # Store the unique agencies that we see
+        unique.add(agency)
+
+        clean_federal.append([domain, agency])
+
+    # Prepare the agency list agency_name : agency_id
+    agency_dict = {}
+    for row in csv.reader(agencies):
+        agency_dict[row[0]] = row[1]
+
+    # Write out the unique agencies
+    for agency in unique:
+        writer.writerow([agency])
+
+    # Create the clean-current-federal for use later.
+    # TODO: Remove this and have it handled in code.
```

Comment:
If this is still valid, you should create an issue for it and reference it here.

Reply:
Please see commit a17deb4.